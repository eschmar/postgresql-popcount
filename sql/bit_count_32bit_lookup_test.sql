CREATE EXTENSION bit_count;

SELECT bit_count_32bit_lookup(15::bit(4));
SELECT bit_count_32bit_lookup(7::bit(4));

SELECT bit_count_32bit_lookup(15::bit(8));
SELECT bit_count_32bit_lookup(7::bit(8));

SELECT bit_count_32bit_lookup(15::bit(12));
SELECT bit_count_32bit_lookup(7::bit(12));

SELECT bit_count_32bit_lookup(15::bit(32));
SELECT bit_count_32bit_lookup(7::bit(32));

SELECT bit_count_32bit_lookup(30::bit(32));
SELECT bit_count_32bit_lookup(127::bit(32));

SELECT sum(bit_count_32bit_lookup(x::bit(512))) from generate_series(1,1000000) x;
SELECT sum(bit_count_32bit_lookup(x::bit(500))) from generate_series(1,1000000) x;
