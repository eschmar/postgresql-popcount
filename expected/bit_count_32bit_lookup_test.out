CREATE EXTENSION bit_count;
ERROR:  extension "bit_count" already exists
SELECT bit_count_32bit_lookup(15::bit(4));
 bit_count_32bit_lookup 
------------------------
                      4
(1 row)

SELECT bit_count_32bit_lookup(7::bit(4));
 bit_count_32bit_lookup 
------------------------
                      3
(1 row)

SELECT bit_count_32bit_lookup(15::bit(8));
 bit_count_32bit_lookup 
------------------------
                      4
(1 row)

SELECT bit_count_32bit_lookup(7::bit(8));
 bit_count_32bit_lookup 
------------------------
                      3
(1 row)

SELECT bit_count_32bit_lookup(15::bit(12));
 bit_count_32bit_lookup 
------------------------
                      4
(1 row)

SELECT bit_count_32bit_lookup(7::bit(12));
 bit_count_32bit_lookup 
------------------------
                      3
(1 row)

SELECT bit_count_32bit_lookup(15::bit(32));
 bit_count_32bit_lookup 
------------------------
                      4
(1 row)

SELECT bit_count_32bit_lookup(7::bit(32));
 bit_count_32bit_lookup 
------------------------
                      3
(1 row)

SELECT bit_count_32bit_lookup(30::bit(32));
 bit_count_32bit_lookup 
------------------------
                      4
(1 row)

SELECT bit_count_32bit_lookup(127::bit(32));
 bit_count_32bit_lookup 
------------------------
                      7
(1 row)

SELECT sum(bit_count_32bit_lookup(x::bit(512))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT sum(bit_count_32bit_lookup(x::bit(500))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

